### PR TITLE
feat: Show/Hide Managed Fields UI from manifest

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

The managedFields config under the manifest takes up a larger view section and should be toggled by help of a checkbox.

Fixes [AB 1972](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=1972)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Reference
reference of managedFields section of a manifest :

image

# Expected UX:
- Hide Managed Fields checkbox to toggle, i.e. to show/hide the managedFields section.
- It should be relevant and only be shown if :
  - Manifest tab is active.
  - ManagedFields exists under the selected manifest.
- On selected checkbox, the managedFields should be removed (not collapsed) from the manifest. 

image

# How Has This Been Tested?

- Go to any app details (must be deployed)
- Select any resource
- Open the manifest for that resource
- If managedFields exist, there should be a checkbox [Hide Managed Fields] to show/hide the managed fields

- [x] Tested for manifests having managedFields
- [x] Tested for manifests not having managedFields

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 1]
```diff
@@ -35,11 +35,19 @@ function NodeDetailComponent({
...
     const [isResourceDeleted, setResourceDeleted] = useState(false)
+    const [isManagedFields, setManagedFields] = useState(false)
+    const [showManagedFields, setShowManagedFields] = useState(false)
     const [fetchingResource, setFetchingResource] = useState(
         isResourceBrowserView && params.nodeType === Nodes.Pod.toLowerCase(),
     )
     const { path, url } = useRouteMatch()
+    const toggleManagedFields = (managedFieldsExist: boolean) => {
+        if (selectedTabName === NodeDetailTab.MANIFEST && managedFieldsExist) setManagedFields(true)
+        else setManagedFields(false)
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:46]: Please use curly braces

Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx]: Added curly braces

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 2]
```diff
@@ -178,6 +189,21 @@ function NodeDetailComponent({
...
                     })}
+                {isManagedFields ? (
+                    <>
+                        <div className="ml-12 mr-5 tab-cell-border"></div>
+                        <div className="pt-6 pb-6 pl-8 pr-8 top">
+                            <Checkbox
+                                rootClassName="mb-0-imp h-20"
+                                isChecked={hideManagedFields}
+                                value={CHECKBOX_VALUE.CHECKED}
+                                onChange={() => setHideManagedFields(!hideManagedFields)}
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:200]: Mitigate inline function calling

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx [hunk 3]
```diff
@@ -130,9 +133,50 @@ function ManifestComponent({
...
+    }, [params.actionName])
+
+    useEffect(() => {
+        if (activeTab === 'Live manifest') {
+            let jsonManifestData = YAML.parse(activeManifestEditorData)
+            if (jsonManifestData?.metadata?.managedFields) {
+                toggleManagedFields(true)
+                if (hideManagedFields) {
+                    setTrimedManifestEditorData(trimManifestData(jsonManifestData))
+                } else {
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx:158]: Please Optimize code here

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx [hunk 4]
```diff
@@ -130,9 +133,50 @@ function ManifestComponent({
...
+            }
+        } else {
+            setTrimedManifestEditorData(activeManifestEditorData)
+        }
+    }, [activeManifestEditorData, hideManagedFields, activeTab])
+
     //For External
+    const trimManifestData = (jsonManifestData: object): string => {
+        const _trimedManifestData = JSON.stringify(jsonManifestData, (key, value) => {
+            if (key === 'metadata') {
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx:172]: Please take it from constant

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx (hunk 3 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx]: Optimised 👍

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx (hunk 4 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx]: extracted 👍

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 5]
```diff
@@ -172,9 +183,11 @@ function NodeDetailComponent({
         selectedResource.containers = resourceContainers
     }
 
+    const handleChanges = () => setHideManagedFields(!hideManagedFields)
```
[the remark this answers is not in the document]
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx]: Added 👍

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 6]
```diff
@@ -38,11 +38,22 @@ function NodeDetailComponent({
     const [selectedTabName, setSelectedTabName] = useState('')
     const [resourceContainers, setResourceContainers] = useState([])
     const [isResourceDeleted, setResourceDeleted] = useState(false)
+    const [isManagedFields, setManagedFields] = useState(false)
+    const [hideManagedFields, setHideManagedFields] = useState(false)
     const [fetchingResource, setFetchingResource] = useState(
         isResourceBrowserView && params.nodeType === Nodes.Pod.toLowerCase(),
     )
     const { path, url } = useRouteMatch()
+    const toggleManagedFields = (managedFieldsExist: boolean) => {
```
Comment (milindk8) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:47]: no need to pass managedFiledsExist inside the function as it is globally available. directly use isManagedFields inside if condition

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 7]
```diff
@@ -172,9 +183,13 @@ function NodeDetailComponent({
         selectedResource.containers = resourceContainers
     }
 
+    const handleChanges = () => {
```
Comment (milindk8) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:186]: function return type missing :void

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx [hunk 8]
```diff
@@ -135,9 +139,45 @@ function ManifestComponent({
         if (!isDeleted && !isEditmode && activeManifestEditorData !== modifiedManifest) {
             setActiveManifestEditorData(modifiedManifest)
         }
+        if (isEditmode) {
+            toggleManagedFields(false)
+            let jsonManifestData = YAML.parse(activeManifestEditorData)
```
Comment (milindk8) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx:144]: replace let with const

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx [hunk 9]
```diff
@@ -135,9 +139,45 @@ function ManifestComponent({
...
+                }
+            }
+        }
+    }, [activeManifestEditorData, hideManagedFields, activeTab])
+
     //For External
+    const trimManifestData = (jsonManifestData: object): string => {
+        const _trimedManifestData = JSON.stringify(jsonManifestData, (key, value) => {
+            if (key === MANIFEST_KEY_FIELDS.METADATA) {
+                value[MANIFEST_KEY_FIELDS.MANAGED_FIELDS] = undefined
```
Comment (milindk8) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx:174]: set the value null

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx (hunk 7 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx]: Implemented 👍

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx (hunk 8 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx]: Implemented 👍

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx (hunk 9 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx]: Implemented 👍

**Review by milindk8 — APPROVED** [review]

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 10]
```diff
@@ -199,6 +214,21 @@ function NodeDetailComponent({
                             </div>
                         )
                     })}
+                {isManagedFields ? (
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:217]: use & syntax like isManagedFields &&
remove the ternary null part

**Review by vivek-devtron — APPROVED** [review]

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx (hunk 9 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Manifest.component.tsx:174]: Reverted back to undefined

**Pull request merged**